src/main/frontend/worker/db/migrate.cljs

```diff
@@ -208,21 +208,21 @@
             ordered-columns-tx
             filters-tx)))
 
-(defn- rename-properties
-  [props-to-rename]
-  (fn [conn _search-db]
+(defn rename-properties
+  [props-to-rename & {:keys [replace-fn]}]
+  (fn [conn]
     (when (ldb/db-based-graph? @conn)
-      (let [props-tx (rename-properties-aux @conn props-to-rename)]
-        ;; Property changes need to be in their own tx for subsequent uses of properties to take effect
```

Comment:
I didn't know we could do this all in one transaction. I thought I had observed in earlier versions of our datascript that we had to define a new property in a separate transaction from transaction data that uses the property. I did see that migration 14 defines :logseq.property.class/extends but that would only apply to really old versions